mongoose.c

```diff
@@ -11011,10 +11013,11 @@ static int mg_tls_server_recv_hello(struct mg_connection *c) {
     MG_INFO(("bad session id len"));
   }
   cipher_suites_len = MG_LOAD_BE16(rio->buf + 44 + session_id_len);
-  if (cipher_suites_len > (rio->len - 46 - session_id_len)) goto fail;
```

Comment:
Is that 46 , and 46 + 4  = 50 be somehow #defined ?

Reply:
All those magic numbers were already there, I just moved them from left to right.
They stop being magic once you have the record structure in your mind, it didn't take me long to get it. That 4 is 2 x BE16, respective length fields being checked

Reply:
46 is 11 + 32 client random + byte length + BE16 length...
11 is fixed header

Reply:
Not quite different from what you did with DHCP... yes, when other dev does it it sends shivers down one's spine, I know.

Reply:
Being these records very dynamic, with many variable lengths, I don't know if being more academic (less hardcoded) would actually help. Maybe we could add some comments and leave those numbers as they are...